File: 03_MOBILE/android/app/src/main/java/com/stratoc/mobile/terminal/transport/TerminalSessionClient.kt
package com.stratoc.mobile.terminal.transport

data class TerminalControlMessage(
    val type: String,
    val reason: String? = null,
    val columns: Int? = null,
    val rows: Int? = null,
)

class TerminalSessionClient {
    var lastConnection: ConnectionRequest? = null
        private set
    var lastResize: Pair<Int, Int>? = null
        private set
    val sentInput = mutableListOf<ByteArray>()

    fun connect(hostUrl: String, token: String, sessionName: String, columns: Int, rows: Int) {
        lastConnection = ConnectionRequest(hostUrl, token, sessionName, columns, rows)
    }

    fun sendInput(bytes: ByteArray) {
        sentInput += bytes
    }

    fun sendResize(columns: Int, rows: Int) {
        lastResize = columns to rows
    }
}

data class ConnectionRequest(
    val hostUrl: String,
    val token: String,
    val sessionName: String,
    val columns: Int,
    val rows: Int,
)

File: 03_MOBILE/ios/TerminalAttach/Transport/TerminalSessionClient.swift
import Foundation

public struct TerminalControlMessage: Codable, Equatable {
  enum CodingKeys: String, CodingKey {
    case type
    case reason
    case columns
    case rows
  }

  public let type: String
  public let reason: String?
  public let columns: UInt16?
  public let rows: UInt16?

  public init(type: String, reason: String? = nil, columns: UInt16? = nil, rows: UInt16? = nil) {
    self.type = type
    self.reason = reason
    self.columns = columns
    self.rows = rows
  }

  public func encode(to encoder: Encoder) throws {
    var container = encoder.container(keyedBy: CodingKeys.self)
    try container.encode(type, forKey: .type)
    try container.encodeIfPresent(reason, forKey: .reason)
    try container.encodeIfPresent(columns, forKey: .columns)
    try container.encodeIfPresent(rows, forKey: .rows)
  }
}

public final class TerminalSessionClient {
  public private(set) var lastConnection: (hostURL: URL, token: String, sessionName: String, columns: Int, rows: Int)?
  public private(set) var sentInput: [Data] = []
  public private(set) var lastResize: (columns: Int, rows: Int)?

  public init() {}

  public func connect(hostURL: URL, token: String, sessionName: String, columns: Int, rows: Int) {
    lastConnection = (hostURL, token, sessionName, columns, rows)
  }

  public func sendInput(_ data: Data) {
    sentInput.append(data)
  }

  public func sendResize(columns: Int, rows: Int) {
    lastResize = (columns, rows)
  }
}
